src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.NavigateTo
+{
+    internal abstract partial class AbstractNavigateToSearchService : INavigateToSearchService
```

Comment:
Annoyingly, you can only ExportLangaugeService to one language.  And, you cannot put multiple ExportLanguageService attributes on a single class.  So this is how we do it.  We have the base calss with 100% of the functionality, and we have dummy derived classes purely for exporting.  Perhaps @mattwar can take a look and see if we can make it so that ExportLanguageServiceAttribute could be augmented to allow exporting to multiple languages.  That would help a handful of services like this one.